service/history/historyEngine.go

```diff
@@ -293,11 +294,26 @@ func NewEngineWithShardContext(
 	replicationTaskExecutors := make(map[string]replicationTaskExecutor)
 	for _, replicationTaskFetcher := range replicationTaskFetchers.GetFetchers() {
 		sourceCluster := replicationTaskFetcher.GetSourceCluster()
+		// Intentionally use the raw client to create its own retry policy
+		adminClient := shard.GetService().GetClientBean().GetRemoteAdminClient(sourceCluster)
+		adminRetryableClient := admin.NewRetryableClient(
+			adminClient,
+			common.CreateReplicationServiceBusyRetryPolicy(),
+			common.IsResourceExhausted,
+		)
+		// Intentionally use the raw client to create its own retry policy
+		historyClient := shard.GetService().GetClientBean().GetHistoryClient()
+		historyRetryableClient := history.NewRetryableClient(
+			historyClient,
+			common.CreateReplicationServiceBusyRetryPolicy(),
+			common.IsResourceExhausted,
+		)
 		nDCHistoryResender := xdc.NewNDCHistoryResender(
 			shard.GetNamespaceCache(),
-			shard.GetService().GetClientBean().GetRemoteAdminClient(currentClusterName),
+			adminRetryableClient,
 			func(ctx context.Context, request *historyservice.ReplicateEventsV2Request) error {
-				return historyEngImpl.ReplicateEventsV2(ctx, request)
+				_, err := historyRetryableClient.ReplicateEventsV2(ctx, request)
+				return err
```

Comment:
here, the logic should target directly at the local shard (this shard).

my guess, the idea here is to wrap the ReplicateEventsV2 API call (i.e. historyRetryableClient) for retry
the only easy change is to use the retryable client, but the solution involves network stack (less efficiency)

let us cut a ticket to unify the client / server interface and update the logic here